src/components/form/Editor.js

```diff
@@ -403,10 +403,13 @@ class Editor extends Component {
             if (typeof dataIndices[CommonConstants.GT_ROW_ID] !== CommonConstants.UNDEFINED) {
                 delete dataIndices[CommonConstants.GT_ROW_ID];
             }
-            fetch(ajaxUrl, {
-                method: settings.method,
-                body: JSON.stringify(dataIndices)
-            }).then(response => response.json()).then((data) => {
+            (settings.method == 'PROMISE'?
+                ajaxUrl(dataIndices) :
```

Comment:
Here u want to get json data to prepare it before sending via ajax I suppose. Can u show an example from the settings? So I can set it on my side and test?

Reply:
postEdit = (data) => {
		return new Promise( (resolve) => {
			console.log('post edit:',data);
                        callAFunctionUsingSomeRestAPI(data).then( () => {
			resolve({json:function(){return data;}});
                         }
			});

For testing proposes callAFunctionUsingSomeRestAPI can be Promise.resolve()

Then in edit configuration:

	    ajax: {
			edit: {
				 url:that.postEdit,
				 type: 'PROMISE'
			 }
		 },

Reply:
I've done research and concluded, that this approach is a bad practice in terms of "open/closed" principle in SOLID. For instance, any malware, developer or user can substitute the `ajaxUrl()` with any function they need and execute any inner plug-in`s functions they need, with any parameters (XSS attack) - that is, it breaks everything. So, I'm sorry, but this is the security problem. Also, this is not a good way to implement different function in HTTP1.1 interface, because PROMISE is not a HTTP1.1 method implementation. 
If you can provide some arguments and/or rework your PR, we can proceed to work on it.